sdk/metric/internal/aggregate/sum.go

```diff
@@ -6,58 +6,92 @@ package aggregate // import "go.opentelemetry.io/otel/sdk/metric/internal/aggreg
 import (
 	"context"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/sdk/metric/metricdata"
 )
 
 type sumValue[N int64 | float64] struct {
-	n     N
+	n     atomicCounter[N]
 	res   FilteredExemplarReservoir[N]
 	attrs attribute.Set
 }
 
 // valueMap is the storage for sums.
 type valueMap[N int64 | float64] struct {
-	sync.Mutex
-	newRes func(attribute.Set) FilteredExemplarReservoir[N]
-	limit  limiter[sumValue[N]]
-	values map[attribute.Distinct]sumValue[N]
+	newRes   func(attribute.Set) FilteredExemplarReservoir[N]
+	aggLimit int
+
+	// cumulative sums do not reset values during collection, so in that case
+	// clearValuesOnCollection is false, hcwg is unused, and only values[0]
+	// and len[0] are used. All other aggregations reset on collection, so we
+	// use hcwg to swap between the hot and cold maps and len so measurements
+	// can continue without blocking on collection.
+	//
+	// see hotColdWaitGroup for how this works.
+	clearValuesOnCollection bool
+	hcwg                    hotColdWaitGroup
+	values                  [2]sync.Map
+	len                     [2]atomic.Int64
```

Comment:
Splitting this into a type dedicated to cumulative values seems appropriate. It would have fields like this right?

```suggestion
	values                  sync.Map
	len                     atomic.Int64
```

Reply:
Done in https://github.com/open-telemetry/opentelemetry-go/pull/7427/commits/b71918beccf18f34b3bdab83a2d8eb878eeee66e